### ISSUE TITLE
Module - output a list directly with the splat ('*') operator

### ISSUE THREAD

**Issue by ocsi01** [hashicorp/terraform#7430]
I'm creating a list of subnets in my module:

```
resource "aws_subnet" "public" {
  vpc_id            = "${var.vpc_id}"
  cidr_block        = "${element( var.cidrs, count.index)}"
  availability_zone = "${element(var.azs, count.index)}"
  count             = "${length(var.cidrs)}"

  tags      { Name = "${var.name}.${element(var.azs, count.index)}" }
  lifecycle { create_before_destroy = true }

  map_public_ip_on_launch = true
}
```

Setting the output as a list of the created subnet ids:

`output "subnet_ids" { value = "${aws_subnet.public.*.id}" }
`
The error is:

> output 'subnet_ids': use of the splat ('*') operator must be wrapped in a list declaration

Workaround:

`output "subnet_ids" { value = "${split(",",join(",", aws_subnet.public.*.id))}" }
`
### Terraform Version
Terraform v0.7.0-rc2 (46a0709bba004d8b6e0eedad411270b3ae135a9e)

### Affected Resource(s)
- splat ('*') operator
- lists
- outputs
- core?

### Expected Behavior
The use of splat should be directly supported without hacking. ( It is already an array.)

### Actual Behavior
I believe the splat operator is not aware of the fact, an output can be an array.

**Comment by jen20** [comment]
Hi @ocsi01! Thanks for reporting this - it's actually a case of documentation lagging code. I assume from this that you are using a release candidate of Terraform 0.7 or a build from master. The correct output definition here is:

```hcl
output "subnet_ids" {
    value = ["${aws_subnet.public.*.id}"]
} 
```

I do think it might be possible to expand these automatically without the `[]` wrapper - I'll open a separate issue for that for discussion - it would bring more consistency.

I'll go ahead and close this issue - if the fix proposed doesn't work for you please feel free to re-open!

**Comment by laura-herrera** [comment]
Hi, I've tried exactly that and doesn't work:
Terraform v0.10.7
output "private_subnets" {
  value = ["${aws_subnet.private.*.id}"]
}

$ terraform plan
module 'cluster': aws_subnet.private.*.id is not a valid output for module network
please advise

**Comment by MichalTaratuta** [comment]
This is still broken

**Comment by MichalTaratuta** [comment]
Resolved with updateding to Terraform v0.10.8
